src/GrumPHP/Locator/ChangedFiles.php

```diff
@@ -41,8 +42,20 @@ public function __construct(Repository $repository, Filesystem $filesystem)
     public function locateFromGitRepository()
     {
         $diff = $this->repository->getWorkingCopy()->getDiffStaged();
+//      if($context instanceof GitPrePushContext ) {
+        $actualbranch = trim(shell_exec('git name-rev --name-only HEAD'));
+        $diff = explode("\n", trim(shell_exec('git diff --name-only origin/' . $actualbranch . '..HEAD --oneline')));
```

Comment:
As mentioned above: what if the remote branch does not exist? Maybe it's better to use a commit hash instead of the parsed branch name.